packages/nimble-components/src/wafer-map/specs/features/rendering.md

```diff
@@ -82,46 +81,48 @@ The POC is found in this branch [Worker Rendering POC](https://github.com/ni/nim
+The best solution to solve the API of the wafermap is to use Apache Arrow as the wafer component API, and Typed Arrays as the worker API for their iterating performance and transferability to worker threads.
+
+The Public API will be the following:
 
 ```TS
-class WaferData {
-    // the x coordinates of each column of dies
-    dieColIndexArray: Int32Array;
-    // the lengths of each row of dies
-    rowLengthsArray: Int32Array;
-    // the y coordinates of each die as a matrix row by row
-    dieRowIndexLayer: Int32Array;
-    // the value of each die as a matrix row by row
-    dieValuesLayer: Float64Array;
-    // the highlight  approach is still undecided, we have two options:
-    // the highlight state of each die as a matrix; user will have to pre-calculate tags into highlighted conditions.
-    dieHighlightsLayer: Int8Array;
-    // a 32 bitset array of tags for each die; aligns more closely to the existing public api but limits users to 32 tags.
-    dieHighlightsLayer: Int32Array;
-    // metadata array for each die; it will not be sent to the worker
-    metadata : unknown[]
+import { Table, TypeMap } from 'apache-arrow';
+
+export interface WaferMapTableType extends TypeMap {
+    colIndex: Int32;
+    rowIndex: Int32;
+    value: Float32;
+}
+
+export class WaferMap<T extends WaferMapTableType> extends FoundationElement {
+...
+public diesTable: Table<T> | undefined;
+public highlightedTable: Table<T> | undefined;
+...
 }
 ```
 
-Using TypedArrays has the benefit of direct transfer to web workers without structured cloning of the object by transferring the arrayBuffers and reconstructing the object. Other benefits of typedArrays include the low access time when iterating over the values, more memory efficiency and faster direct access to metadata layers values. The previous inputs can be adapted to this new structure to maintain backwards compatibility.
+This will be the [Apache Arrow](https://arrow.apache.org/docs/js/classes/Arrow_dom.Table.html) table schema.
+It will require at least three columns for the `diesTable`:
 
-This API will have [optimized byte-array interop from Blazor](https://learn.microsoft.com/en-us/dotnet/core/compatibility/aspnet-core/6.0/byte-array-interop) and should be supported by Angular as a [vanilla javascript feature](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/ArrayBuffer).
+-   The row and column indices will be `Int32` columns
+-   The values will be a `Float32` column.
 
-The alternative to the above mentioned data structure is an [apache arrow](https://arrow.apache.org/docs/js/index.html) table with columns and metadata.
+If there are more columns needed to store metadata or other values the schema will be extensible.
 
-Pros of using Apache Arrow:
+The `highlightedTable` will contain rows partially filled with values which will be used to filter the `diesTable` and enable highlighting.
```

Comment:
What would the `value` field mean in the `highlightedTable` data structure? 
1. Is it ignored, with any die that appears in `highlightedTable` being highlighted?
2. Is it treated as a boolean, where dies with value 0 are unhighighlighted and value >0 are highlighted?
3. Does it have some other meaning, like each value getting a different highlight color or being included in metadata?

If it's 1 or 2 we could consider changing its type to capture this more accurately. That might mean making the class generic on two types, `TDieValues` and `THighlightValues` or something like that.

We should document this in more detail here.